From 2f31a5e87c4d664da68545ef5b3e63c09be7cca8 Mon Sep 17 00:00:00 2001
From: Prakash.Kumar <prakumar@habana.ai>
Date: Mon, 12 Oct 2020 16:51:49 +0300
Subject: [PATCH] [SW-21029] goya2_arc: add support for vuart logging

Added a function which facilitates the logging on
vUART. This change has been tested with hl-smi
dmon.

Change-Id: Ia7c903bc85d2535d89eef5aa847a425cb267c4ee
Signed-off-by: Prakash.Kumar <prakumar@habana.ai>
---

diff --git a/goya2_arc/common.c b/goya2_arc/common.c
index 5ad8c4a..6ba72f5 100644
--- a/goya2_arc/common.c
+++ b/goya2_arc/common.c
@@ -32,6 +32,9 @@
 #define FW_WORD_NUM		0x2200	/* FW_SIZE / 2 */
 #define PRODUCT_ID_STR_SIZE	0x8
 
+#define mmPCIE_WRAP_VUART_TX_DATA       0xC15F60
+#define mmPCIE_WRAP_VUART_TX_STATUS     0xC15F64
+
 static void preboot_init_pcie_phy_fw_update(void)
 {
 	/* Configure FW update size */
@@ -201,7 +204,17 @@
 }
 
 void preboot_delay(void) {}
-void preboot_putc_virt_uart(const char c) {}
+
+void preboot_putc_virt_uart(const char c)
+{
+	/*
+	 * update vUART to HOST
+	 * DATA: mmPCIE_WRAP_VUART_TX_DATA
+	 * STATUS: mmPCIE_WRAP_VUART_TX_STATUS
+	 */
+	regmap_writel(c, mmPCIE_WRAP_VUART_TX_DATA);
+	regmap_writel(1, mmPCIE_WRAP_VUART_TX_STATUS);
+}
 
 u32 get_cs_pad_sel(void)
 {
